docs/docs/unit-testing.md

```diff
@@ -0,0 +1,257 @@
+with a minimal config.
+
+```js
+// jest-preprocess.js
+const babelOptions = {
+  presets: ["@babel/react", "@babel/env"],
+  plugins: [
+    "@babel/plugin-proposal-optional-chaining",
+    "@babel/plugin-proposal-class-properties",
+  ],
+}
+
+module.exports = require("babel-jest").createTransformer(babelOptions)
+```
+
+Back to the Jest config, you can see the next option is `testRegex`. This is the
+pattern telling Jest which files contain tests. The pattern above matches any
+`.js` file inside a `__tests__` directory, or any file elsewhere with the
+extension `.test.js` or `.spec.js`. You are telling Jest to ignore any tests in
+the `node_modules` or `.cache` directories.
+
+The `moduleNameMapper` section works a bit like webpack rules, and tells Jest
+how to handle imports. You are mainly concerned here with mocking static file
+imports, which Jest can't handle. A mock is a dummy module that is used instead
+of the real module inside tests. It is good when you have something that you
+can't or don't want to test. You have two types that you mock: stylesheets, and
+everything else. For stylesheets you need to use the package
+`identity-obj-proxy`. For all other files you need to use a manual mock called
+`fileMock.js`. You need to create this yourself. The convention is to create a
+directory called `__mocks__` in the root directory for this. Note the pair of
+double underscores in the name.
+
+```js
+// __mocks__/fileMock.js
+module.exports = "test-file-stub"
+```
+
+The final config setting is `transformIgnorePatterns`. This is very important,
+and is different from what you'll find in other Jest guides. The reason that you
+need this is because Gastby includes un-transpiled ES6 code. By default Jest
+doesn't try to transform code inside `node_modules`, so you will get an error
+like this:
+
+```
+/my-blog/node_modules/gatsby/cache-dir/gatsby-browser-entry.js:1
+({"Object.<anonymous>":function(module,exports,require,__dirname,__filename,global,jest){import React from "react"
+                                                                                            ^^^^^^
+SyntaxError: Unexpected token import
+```
+
+This is because `gatsby-browser-entry.js` isn't being transpiled before running
+in Jest. You can fix this by changing the default `transformIgnorePatterns` to
+exclude the `gatsby` module.
+
+## Writing tests
+
+A full guide to unit testing is beyond the scope of this guide, but you can
+start with a simple snapshot test to check that everything is working.
+
+First, create the test file. You can either put these in a `__tests__`
+directory, or put them elsewhere (usually next to the component itself), with
+the extension `.spec.js` or `.test.js`. The decision comes down to your own
+taste. For this guide you will be testing the `<Bio />` component, so create a
+`Bio.test.js` file next to it in `src/components`:
+
+```js
+import React from "react"
+
```

Comment:
Spacing here should be consistent with the code further down. L230. No new line between package imports